org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/models/JythonGeneratorModel.java

```diff
@@ -0,0 +1,139 @@
+package org.eclipse.scanning.api.points.models;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.scanning.api.annotation.ui.FieldDescriptor;
+import org.eclipse.scanning.api.annotation.ui.FileType;
+
+/**
+ * 
+ * This class allows loading of a 
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public class JythonGeneratorModel extends AbstractPointsModel {
+
+	@FieldDescriptor(label="Module Name", 
+			         hint="The name of the module to load.\nUsually this is the same as the python file without an ending '.py'", 
+			         fieldPosition=0)
+	private String moduleName;
+	
+	@FieldDescriptor(label="Module Path",
+			         hint="The file path to the module folder.", 
+			         file=FileType.EXISTING_FOLDER,
+			         fieldPosition=1)
+	private String path;
+	
+	@FieldDescriptor(label="Class Name",
+	         hint="The name of the class implementing ScanPointIterator.\nIt must extend Iterator<IPosition> and have three methods:\n size(), getShape(), getRank()", 
+	         fieldPosition=2)
+	private String className;
+
+	@FieldDescriptor(visible=false) // TODO We should probably produce a TypeEditor to allow the arguments to be edited...
+	private List<JythonArgument> arguments;
+
+	@FieldDescriptor(visible=false) // TODO We should probably produce a TypeEditor to allow the keywords to be edited...
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)